### DISCUSSION TITLE
postgresql 10 -> 15

### DISCUSSION THREAD

**Discussion by uRhaineWork** [netbox-community/netbox#11369] [Q&A]
As of netbox 3.4 requires postgresql to be in minimal version of 11. Probably many users just install netbox with standard options, not knowing default is 10, or was not that long time ago. At least I think it is, because my fresh alma/netbox installl was from march 2022.
Does anyone have a decent tried out on netbox tutorial on how to upgrade sql from 10 to 14 or 15? or 13? I've read that v above 13 need some depositories added. 
Version of linux is Alma 8.5.

**Comment by mtinberg** [comment]
I don't know offhand but I believe the PostgreSQL website has extensive documentation including upgrade procedures, they used to require you to pg_dump and pg_restore the database to migrate between versions (which shouldn't take long, even the largest Netbox databases are pretty small, and it's good to have the backup in case something goes awry) but I think newer versions can upgrade the database files on disk so upgrades are even easier.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: uRhaineWork ***@***.***>
Sent: Tuesday, January 3, 2023 8:34 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] postgresql 10 -> 15 (Discussion #11369)


As of netbox 3.4 requires postgresql to be in minimal version of 11. Probably many users just install netbox with standard options, not knowing default is 10, or was not that long time ago. At least I think it is, because my fresh alma/netbox installl was from march 2022.
Does anyone have a decent tried out on netbox tutorial on how to upgrade sql from 10 to 14 or 15? or 13? I've read that v above 13 need some depositories added.
Version of linux is Alma 8.5.

**Comment by NaClnPepper** [comment]
I went from 3.3.0 to 3.4.1 on our test VM (Oracle Linux Server 8.3). PostgreSQL 10 to 13 with no problems following their pg_upgrade procedure.
